salt/cloud/clouds/vmware.py

```diff
@@ -2046,8 +2048,14 @@ def create(vm_):
     )
 
     if 'clonefrom' in vm_:
+        # If datacenter is specified, set the container reference to start search from it instead
+        container_ref = None
+        if datacenter:
+            datacenter_ref = salt.utils.vmware.get_mor_by_property(_get_si(), vim.Datacenter, datacenter)
+            container_ref = datacenter_ref if datacenter_ref else None
+
         # Clone VM/template from specified VM/template
-        object_ref = salt.utils.vmware.get_mor_by_property(_get_si(), vim.VirtualMachine, vm_['clonefrom'])
+        object_ref = salt.utils.vmware.get_mor_by_property(_get_si(), vim.VirtualMachine, vm_['clonefrom'], container_ref)
```

Comment:
The reason behind the error is that you use position based parameter rather than a name based one here, the get_mor_by_property functions' 4th parameter is property_value, if you use here container_ref=container_ref instead it will work.